src/nav/MainStackScreen.js

```diff
@@ -87,7 +88,10 @@ const Stack = createStackNavigator<
   MainStackNavigationProp<>,
 >();
 
-type Props = $ReadOnly<{||}>;
+type Props = $ReadOnly<{|
+  navigation: RootStackNavigationProp<'main-stack'>,
+  route: RouteProp<'main-stack', void>,
+|}>;
 
 export default function MainStackScreen(props: Props) {
```

Comment:
Same here about a comment — now that we have a more complicated hierarchy, it'd be good to document it more.